Client/Frontend/Browser/ContextMenuHelper.swift

```diff
@@ -32,20 +30,57 @@ class ContextMenuHelper: NSObject, TabHelper, UIGestureRecognizerDelegate {
 
     required init(tab: Tab) {
         super.init()
+
         self.tab = tab
 
         let path = Bundle.main.path(forResource: "ContextMenu", ofType: "js")!
         let source = try! NSString(contentsOfFile: path, encoding: String.Encoding.utf8.rawValue) as String
         let userScript = WKUserScript(source: source, injectionTime: WKUserScriptInjectionTime.atDocumentEnd, forMainFrameOnly: false)
         tab.webView!.configuration.userContentController.addUserScript(userScript)
 
+        // Disable the native long-press gesture recognizer to prevent the native WKWebView context
+        // menu from appearing.
+        if let nativeLongPressRecognizer = gestureRecognizerWithDescriptionFragment("action=_longPressRecognized:") {
+            nativeLongPressRecognizer.isEnabled = false
+        }
+
         // Add a gesture recognizer that disables the built-in context menu gesture recognizer.
         // This works by making wkwebview's longpress gestures pass through our gestureRecognizer first.
         // We have to allow textselection gestures to pass through while stopping long press of links.
         gestureRecognizer.delegate = self
         tab.webView!.addGestureRecognizer(gestureRecognizer)
     }
 
+    func gestureRecognizerWithDescriptionFragment(_ descriptionFragment: String) -> UIGestureRecognizer? {
+        guard let tab = self.tab else {
+            return nil
+        }
+
+        guard let webView = tab.webView else {
+            return nil
+        }
+
+        for subview in webView.scrollView.subviews {
+            guard let nativeRecognizers = subview.gestureRecognizers else {
+                continue
+            }
+
+            if let matchingRecognizer = nativeRecognizers.find({ recognizer -> Bool in
+                recognizer.description.contains(descriptionFragment)
+            }) {
+                return matchingRecognizer
+            }
+        }
+
+        return nil
```

Comment:
haha. This is up to you as it is just a style thing. But all these loops are just begging to be turned into a functional style. something like 
`return tab?.webView?.scrollView.subviews.flatMap({ $0.gestureRecognizers }).reduce([], { $0 + $1 }).first(where: { $0.description.contains(descriptionFragment) })`

would work pretty cool. But it is just a style thing. So I dont mind either.

Reply:
haha.. i knew you were gonna say this.. ok